roles/maintenance_32_apache/tasks/main.yml

```diff
@@ -0,0 +1,269 @@
+                  '/var/log/{{ apache_binary_name }}/debug.log']
+                }}"
+  changed_when: no
+  ignore_errors: true
+
+# Find all included .pem, .crt, .key, .csr files in all apache configs
+- <<: *task
+  vars:
+    taskid: 32-009
+    name: "Apache: Extract cert paths from apache configs"
+  ansible.builtin.shell:
+    cmd: "grep -rhE '^(SSLCertificateFile|SSLCertificateKeyFile)' {{ item }} | awk '{print $2}'"
+  loop: "{{ apache_included_config_paths }}"
+  register: apache_cert_files_out
+  changed_when: no
+  ignore_errors: true
+  when: "apache_is_ssl_active is defined and apache_is_ssl_active"
+
+- <<: *task
+  vars:
+    taskid: 32-009
+    name: "Apache: Extract cert paths from apache configs"
+  ansible.builtin.set_fact:
+    apache_cert_files: "{{ apache_cert_files_out.results | map(attribute='stdout_lines') | flatten }}"
+  changed_when: no
+  ignore_errors: true
+
+# Check if logrotate is working correctly for apache logs
+# /var/log is also separately checked inside
+# the generic linux role
+- <<: *task
+  vars:
+    taskid: 32-010
+    name: "Apache: Are there files in the apache log dir that are not rotated? | Find large files"
+  ansible.builtin.find:
+    paths: "{{ apache_log_files }}"
+    size: 16m
+    follow: yes
+  register: apache_large_logfiles
+  changed_when: no
+
+# Some of the files produced by the task above are sparse files, most notably /var/log/lastlog
+# Use stat to figure out the actual on-disk size
+- <<: *task
+  vars:
+    taskid: 32-010
+    name: "Apache: Are there files in the apache log dir that are not rotated? | Find large files"
+  ansible.builtin.stat:
+    path: "{{ item }}"
+    get_attributes: no
+    get_checksum: no
+    get_mime: no
+  loop: "{{ apache_large_logfiles.files | map(attribute='path') | list }}"
+  register: apache_large_logfiles_stat
+  changed_when: no
+
+- <<: *task
+  vars:
+    taskid: 32-010
+    name: "Apache: Are there files in the apache log dir that are not rotated? | Find large files"
+    # 4096 blocks equals 16M (assuming 4k logical blocksize, which is to be expected nowadays)
+    apache_large_files: "{{ apache_large_logfiles_stat.results | map(attribute='stat') | selectattr('blocks', 'gt', 4096) | map(attribute='path') | list }}"
+  ansible.builtin.debug:
+    var: "apache_large_files"
+  changed_when: "apache_large_files | length > 0"
+
+# Check the used config file validity
+- <<: *task
+  vars:
+    taskid: 32-012
+    name: "Apache: Check apache configuration validity"
+  ansible.builtin.command: "{{ apache_binary_path }} -t"
+  register: apache_config_test
+  changed_when: "apache_config_test.rc != 0"
+
+# Test if cert is valid in one month
+- <<: *task
+  vars:
+    taskid: 32-014
+    name: Test whether that certificate is valid in a month
+  community.crypto.x509_certificate_info:
+    path: "{{ item }}"
+    valid_at:
+      one_month_later: "+1m"
+  register: apache_cert_info
+  loop: "{{ apache_cert_files }}"
+  when: item.endswith(".crt")
```

Comment:
You can't rely on the file being called `.crt`. At least `.pem` or `.der` are quite common as well.
I think it would be more sensible to only fetch cert files (and not the key files). As fas as I understand it, this would render this `when` check entirely unnecessary.

Reply:
Yea I agree thats a better approach